TFT/src/User/Menu/common.c

```diff
@@ -82,3 +82,18 @@ const ITEM itemExtLenSteps[ITEM_EXT_LEN_NUM] =
 // List for extruder length/distance change steps
 const float extlenSteps[ITEM_EXT_LEN_NUM] = {1.0f, 5.0f, 10.0f, 100.0f, 200.0f};
 
+// Check time elapsed against the time specified in milliseconds for displaying/updating info on screen
+// Use this for timed screen updates in menu loops only
+bool nextScreenUpdate(uint32_t duration)
+{
+  uint32_t curTime = OS_GetTimeMs();
+  if (curTime > (lastTime + duration))
+  {
+    lastTime = curTime;
+    return true;
+  }
+  else
+  {
+    return false;
+  }
+}
```

Comment:
A lower duration will reset lastTime so nextScreenUpdate() will always be false for higher duration query.